### PR TITLE
CORTX-32399: Remove be/btree.* files and related old btree references

### PR DESCRIPTION
**Pull request by kanchan-chaudhari**
Signed-off-by: Kanchan Chaudhari <kanchan.chaudhari@seagate.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: motr/m0crate/crate.c [hunk 1]
```diff
@@ -1208,54 +1210,111 @@ static size_t btree_key_size(const struct cr_btree_key *cbk)
...
-	.ko_ksize   = cr_key_size,
-	.ko_vsize   = cr_val_size,
-	.ko_compare = cr_cmp
-};
+	M0_BE_ALLOC_PTR_SYNC(btree, seg, NULL);
+	M0_ASSERT(rnode_sz > 0 && m0_is_po2(rnode_sz));
+	rnode_sz_shift = __builtin_ffsl(rnode_sz) - 1;
+	M0_BE_ALLOC_ALIGN_ARR_SYNC(rnode, rnode_sz, rnode_sz_shift, seg, NULL);
+	bt = (struct m0_btree_type){ .tt_id = M0_BT_UT_KV_OPS,
+				     .ksize = sizeof (struct m0_fid),
```
Comment (ydb242) [inline motr/m0crate/crate.c:1229]: no space between sizeof and (

File: motr/m0crate/crate.c [hunk 2]
```diff
@@ -1208,54 +1210,111 @@ static size_t btree_key_size(const struct cr_btree_key *cbk)
...
+static struct m0_btree *cr_btree_create(void)
 {
-	return val != NULL ? strlen(val) + 1 : 0;
-}
+	struct m0_btree            *btree;
+	uint32_t                    rnode_sz_shift;
+	int                         rc;;
+	struct m0_btree_type        bt;
+	struct m0_btree_op          b_op = {};
+	struct m0_btree_rec_key_op  keycmp = { .rko_keycmp = cr_cmp, };
```
Comment (ydb242) [inline motr/m0crate/crate.c:1222]: alignment of =

File: motr/m0crate/crate.c [hunk 3]
```diff
@@ -1208,54 +1210,111 @@ static size_t btree_key_size(const struct cr_btree_key *cbk)
...
+					 };
+
+	M0_BTREE_OP_SYNC_WITH_RC(&kv_op, m0_btree_put(tree, &rec, &insert_cb,
+				 &kv_op, NULL));
+}
+
+static int cr_btree_lookup_callback(struct m0_btree_cb *cb,
+				    struct m0_btree_rec *rec)
+{
+	struct m0_btree_rec     *datum = cb->c_datum;
```
Comment (ydb242) [inline motr/m0crate/crate.c:1280]: Extra spaces between variable name and struct name.

File: motr/m0crate/crate.c [hunk 4]
```diff
@@ -1208,54 +1210,111 @@ static size_t btree_key_size(const struct cr_btree_key *cbk)
...
-			  bo_u.u_btree.t_rc);
+	struct m0_btree_op   kv_op        = {};
+	void                *k_ptr = &kv->kv_key.b_addr;
+	void                *v_ptr = &kv->kv_val.b_addr;
+	m0_bcount_t          ksize = kv->kv_key.b_nob;
+	m0_bcount_t          vsize = kv->kv_val.b_nob;
+	struct m0_btree_rec  rec       = {
+			    .r_key.k_data = M0_BUFVEC_INIT_BUF(&k_ptr, &ksize),
+			    .r_val        = M0_BUFVEC_INIT_BUF(&v_ptr, &vsize),
+			};
```
Comment (ydb242) [inline motr/m0crate/crate.c:1298]: alignment of =

File: motr/m0crate/crate.c [hunk 5]
```diff
@@ -1208,54 +1210,111 @@ static size_t btree_key_size(const struct cr_btree_key *cbk)
...
 {
-	M0_BE_OP_SYNC_RET(op, m0_be_btree_delete(tree, NULL, &op, &kv->kv_key),
-			  bo_u.u_btree.t_rc);
+	struct m0_btree_op   kv_op        = {};
+	void                *k_ptr = &kv->kv_key.b_addr;
+	m0_bcount_t          ksize = kv->kv_key.b_nob;
+	struct m0_btree_key  r_key = {
+				  .k_data  = M0_BUFVEC_INIT_BUF(&k_ptr, &ksize),
+				};
+
```
Comment (ydb242) [inline motr/m0crate/crate.c:1315]: alignment of =

File: be/tool/main.c [hunk 6]
```diff
@@ -79,17 +79,17 @@ static const char *betool_help = ""
 "Usage: m0betool be_log_resize <path> <size> \n"
 "\n";
 
-extern void btree_dbg_print(struct m0_be_btree *tree);
+/* extern void btree_dbg_print(struct m0_be_btree *tree); */
```
[the remark this answers is not in the document]
Comment (kanchan-chaudhari) [reply be/tool/main.c:82]: We have discussed and planned to implement this function and eventually enable these calls, which are commented in this PR.

File: btree/btree.c [hunk 7]
```diff
@@ -13041,6 +13045,11 @@ static void ut_btree_crc_persist_test_internal(struct m0_btree_type   *bt,
 	btree_ut_fini();
 }
 
+/** TBD: Implement function to print the BTree. */
+M0_UNUSED static void btree_dbg_print(struct m0_btree btree)
```
[the remark this answers is not in the document]
Comment (kanchan-chaudhari) [reply btree/btree.c:13049]: Refer https://github.com/Seagate/cortx-motr/pull/1941#discussion_r908542980

File: motr/m0crate/crate.c (hunk 4 above)
Comment (kanchan-chaudhari) [reply motr/m0crate/crate.c]: Done.

File: motr/m0crate/crate.c (hunk 5 above)
Comment (kanchan-chaudhari) [reply motr/m0crate/crate.c]: Done.

File: motr/m0crate/crate.c (hunk 2 above)
Comment (kanchan-chaudhari) [reply motr/m0crate/crate.c:1222]: Done.

File: motr/m0crate/crate.c (hunk 1 above)
Comment (kanchan-chaudhari) [reply motr/m0crate/crate.c]: Done.

**Review by shashank-parulekar — APPROVED** [review]

**Review by RadhaGulhane13 — CHANGES_REQUESTED** [review]

File: be/op.h [hunk 8]
```diff
@@ -90,18 +89,6 @@ struct m0_be_op {
 
 	enum m0_be_op_type  bo_utype; /* bo_u type */
 	union {
```
[the remark this answers is not in the document]
Comment (kanchan-chaudhari) [reply be/op.h:91]: After discussion we decided to keep this implementation as is and just remove btree part.

File: motr/m0crate/crate.c [hunk 9]
```diff
@@ -1208,54 +1210,111 @@ static size_t btree_key_size(const struct cr_btree_key *cbk)
...
-						 &kv->kv_val),
-			  bo_u.u_btree.t_rc);
+	struct m0_btree_op   kv_op        = {};
+	void                *k_ptr = &kv->kv_key.b_addr;
+	void                *v_ptr = &kv->kv_val.b_addr;
+	m0_bcount_t          ksize = kv->kv_key.b_nob;
+	m0_bcount_t          vsize = kv->kv_val.b_nob;
+	struct m0_btree_rec  rec       = {
+			    .r_key.k_data = M0_BUFVEC_INIT_BUF(&k_ptr, &ksize),
+			    .r_val        = M0_BUFVEC_INIT_BUF(&v_ptr, &vsize),
```
[the remark this answers is not in the document]
Comment (kanchan-chaudhari) [reply motr/m0crate/crate.c:1297]: Please refer https://github.com/Seagate/cortx-motr/pull/1941#discussion_r908684020

File: motr/m0crate/crate.c [hunk 10]
```diff
@@ -1208,54 +1210,111 @@ static size_t btree_key_size(const struct cr_btree_key *cbk)
...
+		       m0_vec_count(&rec->r_val.ov_vec));
+	return 0;
 }
 
 static void cr_btree_insert(struct m0_key_val *kv)
 {
-	M0_BE_OP_SYNC_RET(op, m0_be_btree_insert(tree, NULL, &op, &kv->kv_key,
-						 &kv->kv_val),
-			  bo_u.u_btree.t_rc);
+	struct m0_btree_op   kv_op        = {};
```
[the remark this answers is not in the document]
Comment (kanchan-chaudhari) [reply motr/m0crate/crate.c:1259]: Done.

File: motr/m0crate/crate.c [hunk 11]
```diff
@@ -1208,54 +1210,111 @@ static size_t btree_key_size(const struct cr_btree_key *cbk)
...
-	.ko_compare = cr_cmp
-};
+	M0_BE_ALLOC_PTR_SYNC(btree, seg, NULL);
+	M0_ASSERT(rnode_sz > 0 && m0_is_po2(rnode_sz));
+	rnode_sz_shift = __builtin_ffsl(rnode_sz) - 1;
+	M0_BE_ALLOC_ALIGN_ARR_SYNC(rnode, rnode_sz, rnode_sz_shift, seg, NULL);
+	bt = (struct m0_btree_type){ .tt_id = M0_BT_UT_KV_OPS,
+				     .ksize = sizeof (struct m0_fid),
+				     .vsize = -1,
+				   };
```
[the remark this answers is not in the document]
Comment (kanchan-chaudhari) [reply motr/m0crate/crate.c]: After verifying, found out that key and value both are of variable size. Hence, set the ksize and vsize to -1.

File: be/ut/main.c [hunk 12]
```diff
@@ -268,8 +265,6 @@ struct m0_ut_suite be_ut = {
 		{ "actrec",                  m0_be_ut_actrec_test             },
 #endif /* __KERNEL__ */
 		{ "list",                    m0_be_ut_list                    },
-		{ "btree-create_destroy",    m0_be_ut_btree_create_destroy    },
```
[the remark this answers is not in the document]
Comment (kanchan-chaudhari) [reply be/ut/main.c:271]: [CORTX-32652](https://jts.seagate.com/browse/CORTX-32652)
: Explore old btree UT code and finalize test needed to port to new btree. This task will be taken separately.

**Pull request merged**